subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6113,10 +6181,10 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 	uint8_t data_len_max;
 	uint8_t *acad = NULL;
 	uint8_t hdr_buf_len;
-	bool dup = false;
 	uint8_t hdr_len;
 	uint8_t *ptr;
 	int8_t rssi;
+	bool drop;
```

Comment:
It looks like `drop` is only used negated (i.e. as `!drop`). In such cases, I think it is better to revert the value, i.e. rename this to `no_drop` or `keep` as avoid the several negations below.